File: EggDropping.c
#include<stdio.h>
#include<conio.h>
#include<limits.h>
//Recursive implementation

int eggdrop(int,int);
int max(int a,int b)
{
	if(a>b)
		return a;
	else
		return b;
}
int eggdrop(int e,int n)
{
	int min,res=0,x;
	//n number of floors.. e number of eggs
	if(n==1||n==0)
		return n;
	if(e==1)
	 return n;
	min=INT_MAX;
	for(x=1;x<=n;x++)
	{
		res= max(eggdrop(e-1,x-1),eggdrop(e,n-x));
		if(res<min)
			min=res;
			
	}
	
	 return min+1;
}

int main()
{
	int n,e,min_trials;
	printf("\n\nEnter the no. of eggs and the number of floors :---   ");
	scanf("%d%d",&e,&n);
	
	printf("\nThe total no. of trials is :--    %d  ",eggdrop(e,n));
	getch();
	return 0;
}

File: EggDropping.java
import java.util.*;
public class Egg_Dropping {
	public static void main(String args[])
	{
		Scanner sc= new Scanner(System.in);
		System.out.println("Enter the number of eggs :---  ");
		int n=sc.nextInt();
		System.out.println("Enter the number of storeys :---  "); 
		int k=sc.nextInt();
		int s=Egg_Dropping.eggdropping(n, k);
		System.out.println("Maximum number of trials :---  "+s);
	}
	
	
	public static int eggdropping(int n,int k)
	{
		int dp[][]=new int[n+1][k+1];
		
		for(int i=1;i<=n;i++)
		{
			dp[i][0]=0;
			dp[i][1]=1;
		}
		for(int j=1;j<=k;j++)
		{
			dp[1][j]=j;
		}
		
		
		for(int i=2;i<=n;i++)
		{
			for(int j=2;j<=k;j++)
			{
				dp[i][j]=99999999;
				for(int x=1;x<=j;x++)
				{
					int res=(1+Math.max(dp[i-1][x-1],dp[i][j-x]));
					if(res<dp[i][j])
						dp[i][j]=res;
					
				}
			}
		}
		return dp[n][k];
		
	}
}
